File: src/#125/solution.js
var isPalindrome = function (s) {
  let strArr = s.split('')
  let right = strArr.length - 1
  let left = 0


  while (left < right) {
    validCodeL = isValidChar(strArr[left].charCodeAt(0))
    validCodeR = isValidChar(strArr[right].charCodeAt(0))
    if (validCodeL && validCodeR) {
      if (strArr[left].toLowerCase() === strArr[right].toLowerCase()) {
        left++
        right--
      } else {
        return false
      }
    } else {
      if (!validCodeL) {
        left++
      }
      if (!validCodeR) {
        right--
      }
    }
  }

  return true
}

function isValidChar(c) {
  if ((c >= 65 && c <= 90) || (c >= 97 && c <= 122) || (c >= 48 && c <= 57)) {
    return true
  }
  return false
}

File: src/#125/solution.go
package solution

func isPalindrome(s string) bool {
	strArr := []byte(s)
	left := 0
	right := len(strArr) - 1
	for left < right {
		validCodeL, leftChar := isValidCode(strArr[left])
		validCodeR, rightChar := isValidCode(strArr[right])
		if validCodeL && validCodeR {
			if leftChar==rightChar{
				left++
				right--
			}else{
				return false
			}
		} else {
			if !validCodeL {
				left++
			}
			if !validCodeR {
				right--
			}
		}
	}
	return true
}

func isValidCode(s byte) (bool, byte) {
	if s >= 'A' && s <= 'Z' {
		return true, s + 32
	}
	if (s >= '1' && s <= '9') || (s >= 'a' && s <= 'z') {
		return true, s
	}
	return false, 0
}
